### PR TITLE
add Gin V2 keyboard

### PR DESCRIPTION
**Pull request by oceeean**
## Description

add Gin V2 keyboard

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by oceeean — COMMENTED** [review]
reviewed

File: keyboards/ocean/gin_v2/README.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Gin V2
```
Comment (drashna) [inline keyboards/ocean/gin_v2/README.md:1]: Please rename the file so that it is lower case.

Comment (oceeean) [reply keyboards/ocean/gin_v2/README.md]: done!

**Comment by oceeean** [comment]
i accidentally add another keyboard to this PR, is it ok? 😢

**Comment by oceeean** [comment]
reverted :)

File: keyboards/ocean/gin_v2/config.h [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+ * COLS: AVR pins used for columns, left to right
+ * ROWS: AVR pins used for rows, top to bottom
+ * DIODE_DIRECTION: COL2ROW = COL = Anode (+), ROW = Cathode (-, marked on diode)
+ *                  ROW2COL = ROW = Anode (+), COL = Cathode (-, marked on diode)
+ *
+ */
+/*                        0    1    2    3    4    5    6    7    8    9   10  11  12  13  14*/
+#define MATRIX_ROW_PINS { D1, D0, D4, C6, D7, E6, B4, B5 }
+#define MATRIX_COL_PINS { B6, B2, B3, B1, F7, F6, F5, F4 }
+#define UNUSED_PINS
```
Comment (drashna) [inline keyboards/ocean/gin_v2/config.h:44]:
```suggestion
```

File: keyboards/ocean/gin_v2/rules.mk [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = caterina
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
```
Comment (drashna) [inline keyboards/ocean/gin_v2/rules.mk:10]:
```suggestion
BOOTMAGIC_ENABLE = yes       # Virtual DIP switch configuration
```

File: keyboards/ocean/gin_v2/config.h (hunk 2 above)
Comment (oceeean) [reply keyboards/ocean/gin_v2/config.h]: done

File: keyboards/ocean/gin_v2/rules.mk (hunk 3 above)
Comment (oceeean) [reply keyboards/ocean/gin_v2/rules.mk]: done

File: keyboards/ocean/gin_v2/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    KC_TRNS, KC_TRNS, KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+  ),
+
+  [3] = LAYOUT(
+    KC_TRNS, KC_TRNS, KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+  ),
```
Comment (fauxpark) [inline keyboards/ocean/gin_v2/keymaps/default/keymap.c:45]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/ocean/gin_v2/keymaps/default/keymap.c (hunk 4 above)
Comment (oceeean) [reply keyboards/ocean/gin_v2/keymaps/default/keymap.c]: changed

File: keyboards/ocean/gin_v2/readme.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
+# Gin V2
+
+![Gin V2](https://media.discordapp.net/attachments/406032987243806721/862564309406056468/20210708_131916.jpg?width=1166&height=657)
```
Comment (drashna) [inline keyboards/ocean/gin_v2/readme.md:3]: It would be better to have this posted on a dedicated image site, such as imgur.

Comment (oceeean) [reply keyboards/ocean/gin_v2/readme.md]: changed to imgur, thank you for the suggestion.

File: keyboards/ocean/gin_v2/gin_v2.h [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+) \
+{ \
+  { K00, K01, K02, K03, K04, K05, K06, K07 }, \
+  { K10, K11, K12, K13, K14, K15, K16      }, \
+  { K20, K21, K22, K23, K24, K25, K26, K27 }, \
+  { K30, K31, K32, K33, K34, K35, K36      }, \
+  { K40, K41, K42, K43, K44, K45, K46, K47 }, \
+  { K50, K51, K52, K53, K54, K55, K56      }, \
+  { K60, K61, K62, K63,      K65, K66, K67 }, \
+  { K70, K71, K72,      K74, K75, K76      }, \
```
Comment (fauxpark) [inline keyboards/ocean/gin_v2/gin_v2.h:34]:
```suggestion
  { K00, K01, K02, K03, K04, K05, K06, K07 }, \
  { K10, K11, K12, K13, K14, K15, K16, XXX }, \
  { K20, K21, K22, K23, K24, K25, K26, K27 }, \
  { K30, K31, K32, K33, K34, K35, K36, XXX }, \
  { K40, K41, K42, K43, K44, K45, K46, K47 }, \
  { K50, K51, K52, K53, K54, K55, K56, XXX }, \
  { K60, K61, K62, K63, XXX, K65, K66, K67 }, \
  { K70, K71, K72, XXX, K74, K75, K76, XXX } \
```

File: keyboards/ocean/gin_v2/gin_v2.h [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#pragma once
+
+#include "quantum.h"
```
Comment (fauxpark) [inline keyboards/ocean/gin_v2/gin_v2.h:18]:
```suggestion
#include "quantum.h"

#define XXX KC_NO
```

File: keyboards/ocean/gin_v2/readme.md [hunk 8]
```diff
@@ -0,0 +1,24 @@
+# Gin V2
+
+![Gin V2](https://i.imgur.com/CQiImvu.jpeg)
```
Comment (fauxpark) [inline keyboards/ocean/gin_v2/readme.md:3]:
```suggestion
![Gin V2](https://i.imgur.com/CQiImvuh.jpeg)
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/ocean/gin_v2/readme.md (hunk 8 above)
Comment (oceeean) [reply keyboards/ocean/gin_v2/readme.md:3]: https://i.imgur.com/CQiImvu.jpeg it is already correct

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Comment by oceeean** [comment]
thank you ❤️

**Pull request merged**